awscrt/s3.py

```diff
@@ -545,11 +555,14 @@ def _on_finish(self, error_code, status_code, error_headers, error_body):
 
         error = None
         if error_code:
-            error = awscrt.exceptions.from_code(error_code)
 
+            error = awscrt.exceptions.from_code(error_code)
+            if (error.name == "AWS_ERROR_CRT_CALLBACK_EXCEPTION"
```

Comment:
first check that it's an AwsCrtError. it's possible for this to return a built-in python exception, which doesn't have a `.name` property